app/api/contests/set_additional_time.go

```diff
@@ -183,7 +183,7 @@ func setAdditionalTimeForGroupInContest(
 		JOIN new_expires_at
 			ON new_expires_at.child_group_id = groups_groups.child_group_id
 		SET groups_groups.expires_at = new_expires_at.expires_at
-		WHERE NOW() < groups_groups.expires_at AND groups_groups.parent_group_id = ?`,
+		WHERE groups_groups.parent_group_id = ?`,
```

Comment:
As the previous request is on `ActiveGroupGroups`, is it enough?
And the access expiration from 2 years ago should not be changed, using ` NOW() < new_expires_at.expires_at` seems more appropriate... mmh actually that's not ok for time reductions, so it should be `(NOW() < new_expires_at.expires_at OR NOW() < groups_groups.expires_at)`. Correct?

Reply:
> As the previous request is on ActiveGroupGroups, is it enough?

fixed

Reply:
> And the access expiration from 2 years ago should not be changed, using ` NOW() < new_expires_at.expires_at` seems more appropriate... mmh actually that's not ok for time reductions, so it should be `(NOW() < new_expires_at.expires_at OR NOW() < groups_groups.expires_at)`. Correct?

IMO if we change the additional change, we should update all the affected relations/attempts, it doesn’t matter how old they are. I don’t see the reason to have some custom handling for old expirations: when expires_at changes from -2y to -2y+1h, is it a problem?

Reply:
It is a bit of a problem. First because if we want to understand some end of contest submissions, it is good to know when the user was actually over for the user, this time added afterwards may be misleading (mainly for the attempt, on the perm is less a concern). Second because of the reason of the previous comment, it may cause two attempts to become active while it should not be possible. :-/

Reply:
> As the previous request is on ActiveGroupGroups, is it enough?

No it wouldn't be enough. We want to update only groups_groups for the given (parent_group_id, child_group_id) pair, not all groups_groups with given child_group_id.

Reply:
Fixed both